geonode_mapstore_client/client/js/components/DetailsPanel/DetailsPanel.jsx

```diff
@@ -158,21 +156,15 @@ const MapThumbnailView = ({ initialBbox, layers, onMapThumbnail, onClose, saving
                         ...(layers ? layers : [])
                     ]}
                 >
-                    <FitBounds
-                        mapType="openlayers"
-                        active
-                        geometry={extent}
-                        duration={300}
-                        geometryProjection={projection}
-                    />
+                    <ZoomTo extent={extent} />
                 </Map>
                 {savingThumbnailMap && <div className="gn-details-thumbnail-loader">
                     <Loader size={50} />
                 </div>
                 }
             </div>
-            <div className="gn-detail-extent-action" >
-                <Button className="btn-primary" onClick={() => onMapThumbnail(currentBbox)} ><Message msgId={"gnhome.apply"} /></Button><Button onClick={() => onClose() }><i className="fa fa-close"/></Button></div>
+            <div className="gn-detail-extent-action">
+                <Button className="btn-primary" onClick={() => onMapThumbnail(currentBbox, onClose)} ><Message msgId={"gnhome.apply"} /></Button><Button onClick={() => onClose() }><i className="fa fa-close"/></Button></div>
```

Comment:
I would prefer to avoid to pass an internal callback to an action derived from useState to generate a side effect in the epic.
Instead we should move the internal state to the redux state and connect the action to the deatil plugin

```js
const enableMapThumbnailViewer = (enabled) => ({
    type: ENABLE_MAP_THUMBNAIL_VIEWER,
    enabled
});
```
in this way we  can trigger the action in the epic

```
Observable.of(
    enableMapThumbnailViewer(false);
```